vars/yamlSubstituteVariables.groovy

```diff
@@ -0,0 +1,314 @@
+ */
+@GenerateDocumentation
+void call(Map<String, String> arguments) {
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: arguments) {
+        def script = checkScript(this, arguments)  ?: this
+
+        // load default & individual configuration
+        Map config = ConfigurationHelper.newInstance(this)
+                                        .loadStepDefaults()
+                                        .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+                                        .mixinStageConfig(script.commonPipelineEnvironment, arguments.stageName ?: env.STAGE_NAME, STEP_CONFIG_KEYS)
+                                        .mixin(arguments, PARAMETER_KEYS)
+                                        .use()
+
+        String manifestFilePath = config.manifestFile ?: "manifest.yml"
+        String variablesFilePath = config.variablesFile ?: "manifest-variables.yml"
+        String outputFilePath = config.outputManifestFile ?: manifestFilePath
+
+        // Parameter for tests only. No public API.
+        // This is to skip manifest file deletion in tests, which
+        // would be a nuisance, as it alters the sources repository.
+        Boolean skipDeletion = arguments.skipDeletion ?: false
+
+        File manifestFile = new File(manifestFilePath)
+        File variablesFile = new File(variablesFilePath)
+
+        if (!manifestFile.exists()) {
+            echo "[YamlSubstituteVariables] Could not find YAML file at ${manifestFilePath}. Skipping variable substitution."
+            return
+        }
+
+        if (!variablesFile.exists()) {
+            echo "[YamlSubstituteVariables] Could not find variable substitution file at ${variablesFilePath}. Skipping variable substitution."
+            return
+        }
+
+        def manifestData = null;
+        try {
+            // may return a List<Object>  (in case more YAML segments are in the file)
+            // or a Map<String, Object> in case there is just one segment.
+            manifestData = readYaml file: manifestFilePath
+            echo "[YamlSubstituteVariables] Loaded manifest at ${manifestFilePath}!"
+        }
+        catch(Exception ex) {
+            debug("Exception: ${ex}")
+            error "[YamlSubstituteVariables] Could not load manifest file at ${manifestFilePath}."
+            error "[YamlSubstituteVariables] Exception was: ${ex}"
+        }
+
+        def variablesData = null
+        try {
+            // may return a List<Object>  (in case more YAML segments are in the file)
+            // or a Map<String, Object> in case there is just one segment.
+            variablesData = readYaml file: variablesFilePath
+            echo "[YamlSubstituteVariables] Loaded variables file at ${variablesFilePath}!"
+        }
+        catch(Exception ex) {
+            debug("Exception: ${ex}")
+            error "[YamlSubstituteVariables] Could not load manifest variables file at ${variablesFilePath}."
```

Comment:
See other comment above wrt exception handling.

Reply:
Will fix it. Thanks.

Reply:
**Update** - It's done. I pushed a new commit that resolves this.